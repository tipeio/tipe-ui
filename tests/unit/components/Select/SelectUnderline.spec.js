import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import selectTests from './selectTests'

import TipeSelectUnderline from '@/components/Select/SelectUnderline'

const options = [
  { label: 'Owner', value: 'owner' },
  { label: 'Admin', value: 'admin' }
]

const props = {
  options
}
describe('TipeSelectUnderline.vue', () => {
  it('renders', () => {
    const renderer = createRenderer()
    const wrapper = shallowMount(TipeSelectUnderline, {
      propsData: {
        ...props
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
  it('has correct data-tipe-ui attibute', () => {
    const wrapper = shallowMount(TipeSelectUnderline, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSelectUnderline')
  })
  selectTests(TipeSelectUnderline)
})
