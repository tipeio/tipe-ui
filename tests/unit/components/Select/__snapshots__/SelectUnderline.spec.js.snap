// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipeSelectUnderline.vue renders 1`] = `
<div data-tipe-ui="TipeSelectUnderline" class="select-flex">
  <div tabindex="0" class="select medium">
    <TipeNativeSelect-stub options="[object Object],[object Object]" placeholder="" selectedValue="[object Object]"></TipeNativeSelect-stub>
    <div class="fake-select medium">Owner</div>
    <!---->
  </div>
</div>
`;

exports[`TipeSelectUnderline.vue renders without options 1`] = `
<div data-tipe-ui="TipeSelectUnderline" class="select-flex">
  <div tabindex="0" class="select medium">
    <TipeNativeSelect-stub placeholder="" selectedValue="[object Object]"></TipeNativeSelect-stub>
    <div class="fake-select error medium">Please provide select options</div>
    <!---->
  </div>
</div>
`;
